policy-controller/grpc/src/lib.rs

```diff
@@ -452,3 +604,313 @@ fn convert_filter(filter: InboundFilter) -> proto::http_route::Filter {
+        weight: 1,
+    }
+}
+
+fn default_outbound_http_route(
+    backend: outbound::http_route::WeightedRouteBackend,
+) -> outbound::HttpRoute {
+    let metadata = Some(Metadata {
+        kind: Some(metadata::Kind::Default("default".to_string())),
+    });
+
+    let rules = vec![outbound::http_route::Rule {
+        matches: vec![api::http_route::HttpRouteMatch {
+            path: Some(api::http_route::PathMatch {
+                kind: Some(api::http_route::path_match::Kind::Prefix("/".to_string())),
+            }),
+            ..Default::default()
+        }],
+        backends: Some(outbound::http_route::Distribution {
+            kind: Some(outbound::http_route::distribution::Kind::RandomAvailable(
+                outbound::http_route::distribution::RandomAvailable {
+                    backends: vec![backend],
+                },
+            )),
+        }),
+        filters: Default::default(),
+    }];
+    outbound::HttpRoute {
+        metadata,
+        rules,
+        ..Default::default()
+    }
+}
+
+fn default_balancer_config() -> outbound::backend::balance_p2c::Load {
+    outbound::backend::balance_p2c::Load::PeakEwma(outbound::backend::balance_p2c::PeakEwma {
+        default_rtt: Some(
+            time::Duration::from_millis(30)
+                .try_into()
+                .expect("failed to convert ewma default_rtt to protobuf"),
+        ),
+        decay: Some(
+            time::Duration::from_secs(10)
+                .try_into()
+                .expect("failed to convert ewma decay to protobuf"),
+        ),
+    })
+}
+
+fn default_queue_config() -> outbound::Queue {
+    outbound::Queue {
+        capacity: 100,
+        failfast_timeout: Some(
+            time::Duration::from_secs(3)
+                .try_into()
+                .expect("failed to convert failfast_timeout to protobuf"),
+        ),
+    }
+}
+
+fn convert_tcp_address(ip_addr: IpAddr, port: NonZeroU16) -> TcpAddress {
+    let ip = match ip_addr {
+        IpAddr::V4(ipv4) => Ip::Ipv4(ipv4.into()),
+        IpAddr::V6(ipv6) => {
+            let first = [
+                ipv6.octets()[0],
+                ipv6.octets()[1],
+                ipv6.octets()[2],
+                ipv6.octets()[3],
+                ipv6.octets()[4],
+                ipv6.octets()[5],
+                ipv6.octets()[6],
+                ipv6.octets()[7],
+            ];
+            let last = [
+                ipv6.octets()[8],
+                ipv6.octets()[9],
+                ipv6.octets()[10],
+                ipv6.octets()[11],
+                ipv6.octets()[12],
+                ipv6.octets()[13],
+                ipv6.octets()[14],
+                ipv6.octets()[15],
+            ];
+            Ip::Ipv6(IPv6 {
+                first: u64::from_be_bytes(first),
+                last: u64::from_be_bytes(last),
+            })
+        }
+    };
+    TcpAddress {
+        ip: Some(IpAddress { ip: Some(ip) }),
+        port: port.get().into(),
+    }
+}
```

Comment:
nit: there's [a `From<SocketAddr>` impl for `TcpAddress`](https://docs.rs/linkerd2-proxy-api/latest/linkerd2_proxy_api/net/struct.TcpAddress.html#impl-From%3CSocketAddr%3E-for-TcpAddress), why not just store the weighted addr as a `SocketAddress` and use that?

Reply:
Yes, let's keep all of these conversions in the api crate.